gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/entity/EntityServiceEntityFetcher.java

```diff
@@ -56,6 +59,18 @@ private EntityQueryRequest buildRequest(
     addGroupBys(exploreRequest, builder);
     addSelections(requestContext, exploreRequest, builder);
 
+    // TODO: Push group by down to EQS
+    // If there is a group by, specify a large limit and track actual limit, offset and order by
+    // expression list, so we can compute these once the we get the results.
+    if (requestContext.hasGroupBy()) {
+      // Will need to do the ordering, limit and offset ourselves after we get the group by results
+      builder.setLimit(QueryServiceClient.DEFAULT_QUERY_SERVICE_GROUP_BY_LIMIT);
+      requestContext.setOrderByExpressions(getRequestOrderByExpressions(exploreRequest));
+    } else {
+      // No Group By
+      throw new RuntimeException("Entity request handler only supports group by requests");
```

Comment:
Why? What would fail in this path (other than us not assigning the requested limit and order by)?

Reply:
Currently, the assumption is entity handler will only be used, in case there is a group by (and filters, order bys, and selections) on EDS attributes.

Since, we return a row for every span (_and not a row per entity type_), we can think about extending the support, where we get the attribute value from EDS per entity, and then copy the value across all the rows for the same entity, but there isn't a use case yet

And, I am actually thinking of changing the implementation to return a single row per entity, if the explore context is an entity type, and not return a row for every span

Reply:
Another tidbit here though, if I ask a query like "give me the count of entity ids grouped by some QS attribute", I would expect that count of entity IDs to be based on entity records, not spans. Seems like things are getting fun 😉

Reply:
> I would prioritize aggregation support (with or without group by) over individual record querying.

Aggregation support without group by? What does that mean? 🤔

Reply:
Added support for the above